### PR TITLE
add support for template upload and download.

### PR DESCRIPTION
**Pull request by fthuerk**
This is an initial implementation of the upload/download template features.

### CODE REVIEW

File: nifipy/components.py [hunk 1]
```diff
@@ -24,7 +28,9 @@ class NifiConnection(object):
         BASE: "{url_base}/nifi-api",
         PROCESSOR: "{url_base}/nifi-api/processors/{component_id}",
         CONTROLLER_SERVICE: "{url_base}/nifi-api/controller-services/{component_id}",
-        FLOW: "{url_base}/nifi-api/flow/process-groups/{process_group}/{subpath}"
+        FLOW: "{url_base}/nifi-api/flow/process-groups/{component_id}/",
```
Comment (Condla) [inline nifipy/components.py:31]: @fthuerk I believe this change breaks original functionality. Could you have a look into it?

Time for unit tests... 😺

File: nifipy/components.py [hunk 2]
```diff
@@ -64,9 +70,15 @@ def get_controller_services(self, process_group = "root"):
     def _get(self, url):
         return requests.get(url)
 
-    def _post(self, url, data):
+    def _put(self, url, data):
         return requests.put(url, data=data, headers={"Content-Type": "application/json"})
- 
+
+    def _post(self, url, data):
+        return requests.post(url, data=data, headers={"Content-Type": "application/json", 'Connection': 'keep-alive'})
```
Comment (Condla) [inline nifipy/components.py:62]: I'm not super strict about this, but it's nice to have a look at PEP-8 in general.
Specifically in this case: Max length of a line should be 79 characters (https://www.python.org/dev/peps/pep-0008/#maximum-line-length)

**Review by Condla — CHANGES_REQUESTED** [review]
As discussed :)

File: nifipy/components.py (hunk 1 above)
Comment (fthuerk) [reply nifipy/components.py:31]: This should be fixed now. Exept for unit tests :D

File: nifipy/components.py (hunk 2 above)
Comment (fthuerk) [reply nifipy/components.py:62]: I will consider this in the future. Seems like there are several violations of this style.

**Comment by fthuerk** [comment]
I introduced a Flow class to implement the previous functionality without sub-paths. It seems that especially for the flow API, there are numerous variants and endpoints.. So maybe my endpoint definition is also not the most suitable.

**Pull request merged**